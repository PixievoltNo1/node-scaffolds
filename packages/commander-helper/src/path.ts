import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
