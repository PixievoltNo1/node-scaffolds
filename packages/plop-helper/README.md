<header>
  <h1 align="center">
    <a href="https://github.com/guanghechen/node-scaffolds/tree/main/packages/plop-helper#readme">@guanghechen/plop-helper</a>
  </h1>
  <div align="center">
    <a href="https://www.npmjs.com/package/@guanghechen/plop-helper">
      <img
        alt="Npm Version"
        src="https://img.shields.io/npm/v/@guanghechen/plop-helper.svg"
      />
    </a>
    <a href="https://www.npmjs.com/package/@guanghechen/plop-helper">
      <img
        alt="Npm Download"
        src="https://img.shields.io/npm/dm/@guanghechen/plop-helper.svg"
      />
    </a>
    <a href="https://www.npmjs.com/package/@guanghechen/plop-helper">
      <img
        alt="Npm License"
        src="https://img.shields.io/npm/l/@guanghechen/plop-helper.svg"
      />
    </a>
    <a href="https://github.com/nodejs/node">
      <img
        alt="Node.js Version"
        src="https://img.shields.io/node/v/@guanghechen/plop-helper"
      />
    </a>
    <a href="https://github.com/prettier/prettier">
      <img
        alt="Code Style: prettier"
        src="https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square"
      />
    </a>
  </div>
</header>
<br/>


A collection of utility functions for plop templates.

## Install

* npm

  ```bash
  npm install --save-dev @guanghechen/plop-helper
  ```

* yarn

  ```bash
  yarn add --dev @guanghechen/plop-helper
  ```

## Usage

This package exposed some [inquirer][] question and answer resolver.

### Npm package

* `NpmPackagePreAnswers`:

  Name                  | Type      | Description
  :--------------------:|:---------:|:-----------------:
  `cwd`                 | `string`  | Current workspace directory
  `isMonorepo`          | `boolean` | Whether if this package under a monorepo

* `NpmPackagePromptsAnswers`:

  Name                  | Type      | Description
  :--------------------:|:---------:|:-----------------:
  `packageName`         | `string`  | Npm package name
  `packageAuthor`       | `string`  | Package author
  `packageVersion`      | `string`  | Package version
  `packageDescription`  | `string`  | Package description
  `packageLocation`     | `string`  | Package location (path relative to the current directory)

* `NpmPackageData`:

  Extended `NpmPackagePreAnswers` and `NpmPackagePromptsAnswers` With following
  Additional properties.

  Name                  | Type      | Description
  :--------------------:|:---------:|:-----------------:
  `packageUsage`        | `string`  | Package usage
  `repositoryName`      | `string`  | Git repository name
  `repositoryHomepage`  | `string`  | Git repository homepage

### Example

I recommend you use the following template directory structure:

```
├── boilerplate/
│   ├── package.json.hbs
│   ├── README.md.hbs
│   ├── rollup.config.js.hbs
│   ├── src
│   │   └── index.ts.hbs
│   ├── tsconfig.json.hbs
│   ├── tsconfig.settings.json.hbs
│   └── tsconfig.src.json.hbs
├── node_modules/
├── cli.js
├── index.js
├── package.json
└── README.md
```

Where the `index.js` exposed a default plop config, such as:

  ```javascript
  const {
    createNpmPackagePrompts,
    resolveNpmPackageAnswers,
    resolveNpmPackagePreAnswers,
  } = require('@guanghechen/plop-helper')
  const path = require('path')
  const manifest = require('./package.json')

  module.exports = function (plop) {
    const preAnswers = resolveNpmPackagePreAnswers()
    const defaultAnswers = { packageVersion: manifest.version }
    const { cwd, isMonorepo } = preAnswers

    plop.setGenerator('ts-package', {
      description: 'create template typescript project',
      prompts: [...createNpmPackagePrompts(preAnswers, defaultAnswers)],
      actions: function (_answers) {
        const answers = resolveNpmPackageAnswers(preAnswers, _answers)
        answers.toolPackageVersion = manifest.version

        const resolveSourcePath = p =>
          path.normalize(path.resolve(__dirname, 'boilerplate', p))
        const resolveTargetPath = p =>
          path.normalize(path.resolve(answers.packageLocation, p))
        const relativePath = path.relative(answers.packageLocation, cwd)

        // Assign resolved data into plop templates.
        Object.assign(_answers, answers)

        return [
          {
            type: 'add',
            path: resolveTargetPath('package.json'),
            templateFile: resolveSourcePath('package.json.hbs'),
          },
          {
            type: 'add',
            path: resolveTargetPath('README.md'),
            templateFile: resolveSourcePath('README.md.hbs'),
          },
          !isMonorepo && {
            type: 'add',
            path: resolveTargetPath('rollup.config.js'),
            templateFile: resolveSourcePath('rollup.config.js.hbs'),
          },
        ].filter(Boolean)
      },
    })
  }
  ```

And the `cli.js` exposed a Node.js CLI script, such as:

  ```javascript
  #! /usr/bin/env node

  const { launch } = require('@guanghechen/plop-helper')
  const path = require('path')

  launch(
    process.argv,
    args => ({
      configPath: args.plopfile || path.join(__dirname, 'index.js'),
    })
  )
  ```
## Related

* [@guanghechen/template-ts-package][]
* [@guanghechen/template-tsx-package][]
* [inquirer][]


[homepage]: https://github.com/guanghechen/node-scaffolds/tree/main/packages/plop-helper#readme
[inquirer]: https://github.com/SBoudrias/Inquirer.js/
[@guanghechen/template-ts-package]: https://github.com/guanghechen/node-scaffolds/tree/main/packages/template-ts-package#readme
[@guanghechen/template-tsx-package]: https://github.com/guanghechen/node-scaffolds/tree/main/packages/template-ts-package#readme
