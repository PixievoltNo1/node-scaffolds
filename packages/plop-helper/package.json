{
  "name": "@guanghechen/plop-helper",
  "version": "1.1.0",
  "description": "A collection of utility functions for plop templates",
  "author": {
    "name": "guanghechen",
    "url": "https://github.com/guanghechen/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/guanghechen/node-scaffolds.git",
    "directory": "packages/plop-helper"
  },
  "homepage": "https://github.com/guanghechen/node-scaffolds/tree/master/packages/plop-helper#readme",
  "keywords": [
    "plop"
  ],
  "main": "lib/cjs/index.js",
  "types": "lib/types/index.d.ts",
  "source": "src/index.ts",
  "license": "MIT",
  "engines": {
    "node": ">= 14.15.0"
  },
  "files": [
    "lib/",
    "!lib/**/*.js.map",
    "!lib/**/*.d.ts.map",
    "package.json",
    "CHANGELOG.md",
    "LICENSE",
    "README.md"
  ],
  "scripts": {
    "build": "cross-env NODE_ENV=production rollup -c ../../rollup.config.js",
    "prebuild": "rimraf lib/",
    "prepublishOnly": "cross-env ROLLUP_SHOULD_SOURCEMAP=false yarn build",
    "test": "cross-env TS_NODE_FILES=true jest --config ../../jest.config.js --rootDir ."
  },
  "dependencies": {
    "@guanghechen/npm-helper": "^1.1.0",
    "@guanghechen/option-helper": "^1.1.0",
    "@types/fs-extra": "^9.0.8",
    "chalk": "^4.1.0",
    "fs-extra": "^9.1.0",
    "minimist": "^1.2.5",
    "mock-stdin": "^1.0.0",
    "node-plop": "^0.26.2",
    "ora": "^5.3.0",
    "plop": "^2.7.4",
    "semver-regex": "^3.1.2"
  },
  "devDependencies": {
    "@guanghechen/jest-helper": "^1.1.0",
    "@guanghechen/rollup-config": "^1.1.0"
  }
}
