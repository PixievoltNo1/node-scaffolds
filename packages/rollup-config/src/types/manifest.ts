import type { IRawEntryItem } from './entry'

export interface IManifestEntryFields {
  /**
   * The default CJS module entry filepath.
   */
  readonly main?: string
  /**
   * The default ESM entry filepath.
   */
  readonly module?: string
  /**
   * The default entry input file(s).
   */
  readonly source: string | string[]
  /**
   * The default types filepath.
   */
  readonly types?: string
  readonly exports?: string | IRawEntryItem | Record<string, IRawEntryItem>
}

export interface IManifestWithDependencies {
  /**
   * Dependency list.
   */
  readonly dependencies?: Record<string, string> | string[]
  /**
   * Optional dependency list.
   */
  readonly optionalDependencies?: Record<string, string> | string[]
  /**
   * Peer dependency list.
   */
  readonly peerDependencies?: Record<string, string> | string[]
}

export interface IManifest extends IManifestEntryFields, IManifestWithDependencies {
  /**
   * Package name.
   */
  readonly name: string
}
