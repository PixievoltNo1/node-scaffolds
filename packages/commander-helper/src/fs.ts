import { isFileSync } from '@guanghechen/file-helper'
import fs from 'fs-extra'
import yaml from 'js-yaml'
import path from 'path'

/**
 * Load configuration file with format .json / .yml / .yaml
 *
 * @param filepath
 * @param logger
 */
export async function loadJsonOrYaml(
  filepath: string,
  encoding = 'utf-8',
): Promise<unknown | never> {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): Promise<string> =>
    fs.readFile(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = await loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = await loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}

/**
 * Load configuration file with format .json / .yml / .yaml  (synchronizing)
 *
 * @param filepath
 * @param logger
 */
export function loadJsonOrYamlSync(
  filepath: string,
  encoding = 'utf-8',
): unknown | never {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): string =>
    fs.readFileSync(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}
