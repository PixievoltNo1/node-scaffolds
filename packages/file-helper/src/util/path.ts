import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(workspace: string, targetPath?: string | null): string {
  if (targetPath == null) return workspace
  const filepath: string = path.resolve(workspace, targetPath)
  return path.normalize(filepath)
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetPath
 */
export function relativeOfWorkspace(workspace: string, targetPath: string): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetPath)
  const filepath: string = path.relative(workspace, absoluteDir)
  return path.normalize(filepath)
}
