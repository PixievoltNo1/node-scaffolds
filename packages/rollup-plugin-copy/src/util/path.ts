import path from 'path'

/**
 * Resolve the source path.
 * @param workspace
 * @param srcPath
 * @returns
 */
export const resolvePath = (workspace: string, srcPath: string): string => {
  const filepath = path.resolve(workspace, srcPath)
  return filepath
}

/**
 * Get the path relative to workspace.
 * @param workspace
 * @param filepath
 * @returns
 */
export const relativePath = (workspace: string, filepath: string): string => {
  const srcPath: string = filepath.startsWith(workspace)
    ? filepath.slice(workspace.length).replace(/^[/\\]/, '')
    : filepath
  return srcPath
}
