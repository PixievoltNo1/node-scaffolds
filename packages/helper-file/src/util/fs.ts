import fs from 'node:fs'
import path from 'node:path'

/**
 * Collect all files under the given directory.
 *
 * @param dir
 * @param predicate
 */
export async function collectAllFiles(
  dir: string,
  predicate?: ((p: string, stat: fs.Stats) => Promise<boolean> | boolean) | null,
): Promise<string[]> {
  const results: string[] = []
  await collect(dir)
  return results

  async function collect(filepath: string): Promise<void> {
    const stat = fs.statSync(filepath)
    if (stat.isDirectory()) {
      const filenames = fs.readdirSync(filepath)
      for (const filename of filenames) {
        const nextFilepath = path.join(filepath, filename)
        await collect(nextFilepath)
      }
      return
    }

    if (stat.isFile()) {
      if (predicate == null || (await predicate(filepath, stat))) {
        results.push(filepath)
      }
    }
  }
}

/**
 * Collect all files under the given directory.   (synchronizing)
 *
 * @param dir
 * @param predicate
 */
export function collectAllFilesSync(
  dir: string,
  predicate?: ((p: string, stat: fs.Stats) => boolean) | null,
): string[] {
  const results: string[] = []
  collect(dir)
  return results

  function collect(filepath: string): void {
    const stat = fs.statSync(filepath)
    if (stat.isDirectory()) {
      const filenames = fs.readdirSync(filepath)
      for (const filename of filenames) {
        const nextFilepath = path.join(filepath, filename)
        collect(nextFilepath)
      }
      return
    }
    if (stat.isFile()) {
      if (predicate == null || predicate(filepath, stat)) {
        results.push(filepath)
      }
    }
  }
}
