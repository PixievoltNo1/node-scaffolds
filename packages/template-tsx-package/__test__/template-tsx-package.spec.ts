import {
  composeStringDesensitizers,
  createFilepathDesensitizer,
  createPackageVersionDesensitizer,
  fileSnapshot,
} from '@guanghechen/jest-helper'
import { runPlopWithMock } from '@guanghechen/plop-helper'
import fs from 'fs-extra'
import path from 'path'
import manifest from '../package.json'

const initialCwd = process.cwd()
const outputDir = path.join(__dirname, 'output')

beforeEach(async function () {
  jest.setTimeout(10000)
  if (!fs.existsSync(outputDir)) fs.mkdirpSync(outputDir)
  process.chdir(outputDir)
})

afterEach(async function () {
  fs.removeSync(outputDir)
  process.chdir(initialCwd)
})

const desensitizers = {
  filepath: createFilepathDesensitizer(__dirname),
  packageVersion: createPackageVersionDesensitizer(
    packageVersion => {
      // eslint-disable-next-line jest/no-standalone-expect
      expect(packageVersion).toEqual(manifest.version)
      return '<LATEST>'
    },
    packageName => /^(@guanghechen\/|version$)/.test(packageName),
  ),
}

describe('tsx-package', function () {
  const templateConfig = path.join(__dirname, '../index.js')

  async function runTest(
    plopBypass: string[],
    mockInputs: string[],
    defaultAnswers: Record<string, unknown>,
    expectedPackageLocation: string,
  ): Promise<void> {
    await runPlopWithMock(
      templateConfig,
      plopBypass,
      mockInputs,
      defaultAnswers,
    )

    const targetDir = path.resolve(expectedPackageLocation)
    fileSnapshot(
      targetDir,
      [
        'src/style/index.styl',
        'src/component.tsx',
        'src/index.tsx',
        'rollup.config.js',
        'tsconfig.json',
        'tsconfig.src.json',
      ],
      desensitizers.filepath,
    )

    fileSnapshot(
      targetDir,
      ['package.json', 'README.md'],
      composeStringDesensitizers(
        desensitizers.filepath,
        desensitizers.packageVersion,
      ),
    )
  }

  test('simple', async function () {
    const defaultAnswers = { nickname: 'jojo', isMonorepo: false }
    await runTest(
      ['@guanghechen/waw'],
      ['', '', 'some descriptions', ''],
      defaultAnswers,
      'packages/waw',
    )
  })
})
