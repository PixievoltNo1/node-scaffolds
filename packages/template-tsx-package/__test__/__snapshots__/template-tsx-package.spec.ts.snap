// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsx-package simple: README.md 1`] = `
"<header>
  <h1 align=\\"center\\">
    <a href=\\"https://github.com/guanghechen/guanghechen/tree/master/packages/waw#readme\\">@guanghechen/waw</a>
  </h1>
  <div align=\\"center\\">
    <a href=\\"https://www.npmjs.com/package/@guanghechen/waw\\">
      <img
        alt=\\"Npm Version\\"
        src=\\"https://img.shields.io/npm/v/@guanghechen/waw.svg\\"
      />
    </a>
    <a href=\\"https://www.npmjs.com/package/@guanghechen/waw\\">
      <img
        alt=\\"Npm Download\\"
        src=\\"https://img.shields.io/npm/dm/@guanghechen/waw.svg\\"
      />
    </a>
    <a href=\\"https://www.npmjs.com/package/@guanghechen/waw\\">
      <img
        alt=\\"Npm License\\"
        src=\\"https://img.shields.io/npm/l/@guanghechen/waw.svg\\"
      />
    </a>
    <a href=\\"#install\\">
      <img
        alt=\\"Module Formats: cjs, esm\\"
        src=\\"https://img.shields.io/badge/module_formats-cjs%2C%20esm-green.svg\\"
      />
    </a>
    <a href=\\"https://github.com/nodejs/node\\">
      <img
        alt=\\"Node.js Version\\"
        src=\\"https://img.shields.io/node/v/@guanghechen/waw\\"
      />
    </a>
    <a href=\\"https://github.com/facebook/react\\">
      <img
        alt=\\"React Version\\"
        src=\\"https://img.shields.io/npm/dependency-version/@guanghechen/waw/peer/react\\"
      />
    </a>
    <a href=\\"https://github.com/prettier/prettier\\">
      <img
        alt=\\"Code Style: prettier\\"
        src=\\"https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square\\"
      />
    </a>
  </div>
</header>
<br/>


Some descriptions.

## Install

* npm

  \`\`\`bash
  npm install --save @guanghechen/waw
  \`\`\`

* yarn

  \`\`\`bash
  yarn add @guanghechen/waw
  \`\`\`

## Usage

  * Used in React project

    \`\`\`tsx
    // index.tsx
    import React from 'react'
    import ReactDOM from 'react-dom'
    import HelloWorld from '@guanghechen/waw'
    import '@guanghechen/waw/lib/esm/index.css'

    ReactDOM.render(
      <HelloWorld />,
      document.getElementById('root')
    )
    \`\`\`


## Related


[homepage]: https://github.com/guanghechen/guanghechen/tree/master/packages/waw#readme
"
`;

exports[`tsx-package simple: package.json 1`] = `
"{
  \\"name\\": \\"@guanghechen/waw\\",
  \\"version\\": \\"<LATEST>\\",
  \\"description\\": \\"Some descriptions\\",
  \\"author\\": {
    \\"name\\": \\"guanghechen\\",
    \\"url\\": \\"https://github.com/guanghechen/\\"
  },
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://github.com/guanghechen/.git\\",
    \\"directory\\": \\"packages/waw\\"
  },
  \\"homepage\\": \\"https://github.com/guanghechen/guanghechen/tree/master/packages/waw#readme\\",
  \\"keywords\\": [
    \\"react component\\",
    \\"stylus\\",
    \\"typescript\\"
  ],
  \\"main\\": \\"lib/cjs/index.js\\",
  \\"module\\": \\"lib/esm/index.js\\",
  \\"types\\": \\"lib/types/index.d.ts\\",
  \\"source\\": \\"src/index.tsx\\",
  \\"license\\": \\"MIT\\",
  \\"engines\\": {
    \\"node\\": \\">=14.15.0\\"
  },
  \\"files\\": [
    \\"lib/\\",
    \\"!lib/**/*.js.map\\",
    \\"!lib/**/*.d.ts.map\\",
    \\"package.json\\",
    \\"CHANGELOG.md\\",
    \\"LICENSE\\",
    \\"README.md\\"
  ],
  \\"scripts\\": {
    \\"start\\": \\"rollup -w -c rollup.config.js\\",
    \\"build\\": \\"cross-env NODE_ENV=production rollup -c rollup.config.js\\",
    \\"postinstall\\": \\"yarn build\\",
    \\"prebuild\\": \\"rimraf lib/\\",
    \\"prepublishOnly\\": \\"cross-env ROLLUP_SHOULD_SOURCEMAP=false yarn build\\",
    \\"test\\": \\"cross-env TS_NODE_FILES=true jest --config ../../jest.config.js --rootDir .\\"
  },
  \\"peerDependencies\\": {
    \\"@types/react\\": \\"^17.0.3\\",
    \\"clsx\\": \\"^1.1.1\\",
    \\"react\\": \\"^17.0.1\\"
  },
  \\"devDependencies\\": {
    \\"@guanghechen/rollup-config-tsx\\": \\"^<LATEST>\\",
    \\"@types/react\\": \\"^17.0.3\\",
    \\"clsx\\": \\"^1.1.1\\",
    \\"react\\": \\"^17.0.1\\",
    \\"rollup\\": \\"2.40.0\\"
  },
  \\"browserslist\\": [
    \\"last 2 versions\\",
    \\"Firefox ESR\\",
    \\"> 1%\\",
    \\"ie >= 11\\"
  ]
}
"
`;

exports[`tsx-package simple: rollup.config.js 1`] = `
"import createRollupConfigs from '@guanghechen/rollup-config-tsx'
import path from 'path'
import manifest from './package.json'

const configs = createRollupConfigs({
  shouldSourceMap: false,
  preprocessOptions: {
    input: ['src/style/index.styl'],
    pluginOptions: {
      multiEntryOptions: {
        exports: false,
      },
      postcssOptions: {
        modules: {
          localsConvention: 'camelCase',
          generateScopedName: 'ghc-[local]',
        },
        postcssUrlOptions: {
          url: 'inline',
          maxSize: 0.5, // 0.5 KB
          assetsPath: path.join(__dirname, 'lib'),
          fallback: 'copy',
          basePath: [path.join(__dirname, 'src')],
          useHash: false,
        },
      },
    },
  },
  manifest,
  pluginOptions: {
    typescriptOptions: { tsconfig: 'tsconfig.src.json' },
    postcssOptions: {
      extract: 'index.css',
      minimize: false,
      sourceMap: false,
      modules: {
        localsConvention: 'camelCase',
        generateScopedName: 'ghc-[local]',
      },
      postcssUrlOptions: {
        url: 'inline',
        maxSize: 0.5, // 0.5 KB
        basePath: [path.join(__dirname, 'src')],
        fallback: function (asset) {
          const url = asset.url.replace(/^[/]assets[/]/, '../assets/')
          return url
        },
      },
    },
  },
})

export default configs
"
`;

exports[`tsx-package simple: src/component.tsx 1`] = `
"import React from 'react'
import classes from './style/index.styl'


export interface WawProps {
  content?: string
}


export function Waw(props: WawProps): React.ReactElement {
  return (
    <div className={ classes.container } >
      {props.content}
    </div>
  )
}


Waw.displayName = 'Waw'
"
`;

exports[`tsx-package simple: src/index.tsx 1`] = `
"import { Waw } from './component'

export * from './component'
export default Waw
"
`;

exports[`tsx-package simple: src/style/index.styl 1`] = `
".container
  display: flex
"
`;

exports[`tsx-package simple: tsconfig.json 1`] = `
"{
  \\"extends\\": \\"../../tsconfig\\",
  \\"compilerOptions\\": {
    \\"lib\\": [\\"esnext\\", \\"dom\\", \\"dom.iterable\\"]
  },
  \\"include\\": [\\"src\\", \\"script\\", \\"__test__\\"]
}
"
`;

exports[`tsx-package simple: tsconfig.src.json 1`] = `
"{
  \\"extends\\": \\"../../tsconfig.settings\\",
  \\"compilerOptions\\": {
    \\"lib\\": [\\"esnext\\", \\"dom\\", \\"dom.iterable\\"],
    \\"declarationDir\\": \\"lib/types\\",
    \\"rootDir\\": \\"src\\"
  },
  \\"include\\": [\\"src\\"]
}
"
`;
